### PR TITLE
(WIP) When I am listing the time entries, I want to make sure they are sorted correctly

### PR DESCRIPTION
**Pull request by seantijmons**

### CODE REVIEW

File: components/search-bar/SearchBar.jsx [hunk 1]
```diff
@@ -3,33 +3,69 @@ import propTypes from 'prop-types';
...
+          {title}
+        </div>
+        <div className="search-bar__divider" />
+        <div className="search-bar__sub-title">
+          {`${itemsLength} ${subtitle}`}
+        </div>
+        <select
+          className={`search-bar__selector ${isFilterVisible ? '' : 'search-bar__selector--disabled'}`}
+          name="client"
+          onChange={handleChange}
```
Comment (naomiaz) [inline components/search-bar/SearchBar.jsx]: You can write your `handleChange()` function directly behind `onChange=`:
`onChange={(event) => this.props.toggleActiveFilter(event.target.value)}`

File: components/search-bar/SearchBar.jsx [hunk 2]
```diff
@@ -3,33 +3,69 @@ import propTypes from 'prop-types';
...
 
-SearchBar.defaultProps = {
-  search: 'Search'
-};
+  componentWillUnmount() {
+    this.props.toggleActiveFilter('');
+  }
+
+  handleChange = (event) => {
+    event.persist();
```
Comment (naomiaz) [inline components/search-bar/SearchBar.jsx]: `handleChange` is not a cancelable event

File: components/search-bar/SearchBar.jsx [hunk 3]
```diff
@@ -3,33 +3,69 @@ import propTypes from 'prop-types';
...
+        <div className="search-bar__divider" />
+        <div className="search-bar__sub-title">
+          {`${itemsLength} ${subtitle}`}
+        </div>
+        <select
+          className={`search-bar__selector ${isFilterVisible ? '' : 'search-bar__selector--disabled'}`}
+          name="client"
+          onChange={handleChange}
+        >
+          <option value="" key="All clients">All clients</option>
```
Comment (boris-arkenaar) [inline components/search-bar/SearchBar.jsx]: Why give it a `key`?

File: ducks/time-entries.js [hunk 4]
```diff
@@ -1,25 +1,60 @@
...
+  itemsLength: 0,
   items: []
 };
 
+const timeEntriesRootSelector = state => state.timeEntries;
+
+export const isFormVisibleSelector = state => state.timeEntries.isFormVisible;
+export const isPostingSelector = state => state.timeEntries.isPosting;
+
+export const numberFilteredItemsSelector = (state, stateEntity) => state[stateEntity]
```
Comment (boris-arkenaar) [inline ducks/time-entries.js]: This function is partly generic and partly specific for time entries. It uses a `stateEntity` to acces any part of the state, but it uses the `activeFilter` from `timeEntries`, and the function is defined in the time entries duck.

Comment (seantijmons) [reply ducks/time-entries.js]: i can make a generic ducks for this but then more can be abstracted out timeEntries, like all toggleFormVisibilities or I can make a ducks for searchBar which only does this calculation?

File: ducks/search-bar.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const SET_ACTIVE_FILTER = 'SET_ACTIVE_FILTER';
+
+export const initialState = {
+  activeFilter: ''
+};
+
+export const searchBarActiveFilterSelector = state => state.searchBar.activeFilter;
```
Comment (boris-arkenaar) [inline ducks/search-bar.js]: You could create a root search bar selector to select `state.searchBar` and use that in this and in the other functions.

File: ducks/search-bar.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+export const numberFilteredItemsSelector = (state, stateEntity) => state[stateEntity]
+  .items.filter(item => !state.searchBar.activeFilter
+  || item.client === state.searchBar.activeFilter).length;
```
Comment (boris-arkenaar) [inline ducks/search-bar.js]: The locations of line breaks in this function makes it quite difficult to see what's going on.

File: ducks/search-bar.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const SET_ACTIVE_FILTER = 'SET_ACTIVE_FILTER';
+
+export const initialState = {
+  activeFilter: ''
+};
+
+export const searchBarActiveFilterSelector = state => state.searchBar.activeFilter;
+
+export const numberFilteredItemsSelector = (state, stateEntity) => state[stateEntity]
```
Comment (boris-arkenaar) [inline ducks/search-bar.js]: I still see this function in the time entries ducks as well. It can be removed from there now, right?

**Pull request merged**